File: src/bin/io/RandomAccessFactory.java
package bin.io;

import java.io.File;
import java.io.IOException;

/**
 * @author Bin
 */
public class RandomAccessFactory {

    public static RandomAccessFile from(RandomAccessData randomAccessData) {
        return new BufferedRandomAccessFile(randomAccessData);
    }

    public static RandomAccessFile from(File file, String mode) throws IOException {
        return new BufferedRandomAccessFile(new RandomAccessDataImpl(file, mode));
    }

    public static RandomAccessFile from(String path, String mode) throws IOException {
        return new BufferedRandomAccessFile(new RandomAccessDataImpl(path, mode));
    }

}

File: Android/app/src/main/java/bin/io/RandomAccessFactory.kt
package bin.io

import org.jetbrains.annotations.Contract
import java.io.File
import java.io.IOException

/**
 * @author Bin
 */
object RandomAccessFactory {
    @JvmStatic
    @Contract("_ -> new")
    fun from(randomAccessData: RandomAccessData): RandomAccessFile {
        return BufferedRandomAccessFile(randomAccessData)
    }

    @JvmStatic
    @Contract("_, _ -> new")
    @Throws(IOException::class)
    fun from(file: File, mode: String): RandomAccessFile {
        return BufferedRandomAccessFile(RandomAccessDataImpl(file, mode))
    }

    @JvmStatic
    @Contract("_, _ -> new")
    @Throws(IOException::class)
    fun from(path: String, mode: String): RandomAccessFile {
        return BufferedRandomAccessFile(RandomAccessDataImpl(path, mode))
    }
}